api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,358 @@
+// These proto definitions follow https://protobuf.dev/programming-guides/style/
+// and recommendations outlined in https://static.sched.com/hosted_files/kccncna17/ad/2017%20CloudNativeCon%20-%20Mod%20gRPC%20Services.pdf
+syntax = "proto3";
+package f5.nginx.agent.api.grpc.mpi.v1;
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+import "common.proto";
+import "file.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+// A service outlining the command and control options for a DataPlane
+service CommandService {
+    // Connects NGINX Agents to the Management Plane agnostic of instance data
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse) {}
+    // Reports on instances and their configurations
+    rpc Status(DataPlaneStatus) returns (google.protobuf.Empty) {}
+    // Reports on instance health
+    rpc Health(DataPlaneHealth) returns (google.protobuf.Empty) {}
+    // A decoupled communication mechanism between the data plane and management plane.
+    rpc Subscribe(stream DataPlaneMessage) returns (stream ManagementPlaneMessage) {}
+}
+
+// The connection request is an intial handshake to establish a connection, sending NGINX Agent instance information
+message ConnectionRequest {
+    // Meta-information associated with a message
+    f5.nginx.agent.api.grpc.mpi.v1.common.MessageRequest message_metadata = 1;
+    // instance information associated with the NGINX Agent
+    Instance agent = 2;
+}
+
+// This contains a series of NGINX Agent configurations
+message AgentConfig { 
+    // Command server settings
+    Command command = 1;
+    // Metrics server settings
+    Metrics metrics = 2;
+    // A series of key/value pairs to add more data to the NGINX Agent instance
+    repeated google.protobuf.Struct labels = 3;
+    // A list of features that the NGINX Agent has
+    repeated string features = 4;
```

Comment:
should this be an enum, free form string makes it hard to synchronize on common set of "features" that's understood by the agent

Reply:
potentially, this can evolve in the implementation

Reply:
an enum makes it harder to extend, right? If I want to add an agent feature that only makes sense for NGINXaaS, the flexibility of `string` is useful.

Reply:
why is it harder to extend? You'll need to add the string value somewhere anyways, since there would need to be agreement on what a particular value "means". So you don't end up with one side thinking it's "fooooo" for one thing, and another side thinking it's "fooo", contrive example, but you get the point.

Reply:
I understand and agree with enums to avoid magic string shenanigans, my concern is about extensibility. 
Maybe I'm missing something about gRPC protos, I don't understand how I can extend the enum in my custom NGINXaaS agent. Say we do use and enum here:

```proto
enum AgentFeature { CONFIG=0, METRICS=1 }
```

Then in my NGINXaaS agent, I have a azure-specific feature that is _not_ part of the shared protocol, say `AZURE_MONITOR`. I don't want to pollute all agents everywhere with my Azure-specific feature, and want freedom to iterate the NGINXaaS agent/server independently of other NGINX products. My agent is in golang, using this protocol directly via `import "github.com/nginx/agent/v3/api/grpc/mpi/v1"`.

Is there a way to add to `enum AgentFeature` after the fact for NGINXaaS?